src/components/Datatable/Datatable.vue

```diff
@@ -0,0 +1,280 @@
+        <th v-if="windowWidth <= 1200" class="mobile-sort">
+          <slot name="header-sort" :handleMobileSort="sortColumns">
+            <Button @click="sortColumns(columns[0])">Sort: A-Z</Button></slot
+          >
+        </th>
+      </tr>
+
+      <tr v-if="windowWidth >= 1200" class="table-row">
+        <th
+          v-for="(column, columnIndex) in columns"
+          class="table-header"
+          :style="{ textAlign: column.align ? column.align : 'start' }"
+          :key="columnIndex"
+          @click="sortColumns(column)"
+        >
+          <div
+            class="icon-container"
+            :style="{ textAlign: column.align ? column.align : 'start' }"
+          >
+            <slot name="column" :column="column">
+              <slot
+                :name="'column-' + columns[columnIndex].value"
+                :column="column"
+              >
+                {{ columns[columnIndex].text }}
+              </slot>
+
+              <i
+                v-if="currentSortDir == 'asc'"
+                class="pi pi-arrow-up-line pi-1x"
+              />
+              <i
+                v-else-if="currentSortDir == 'desc'"
+                class="pi pi-arrow-down-line pi-1x"
+              />
+
+              <div v-else class="rotate-arrow">
+                <i class="pi pi-code-line pi-1x" />
+              </div>
+            </slot>
+          </div>
+        </th>
+      </tr>
+      <tr class="table-row">
+        <td :colspan="Object.keys(columns).length" class="search-input">
+          <Input v-model="search" placeholder="Search...">
+            <template v-slot:prepend>
+              <div class="search-icon">
+                <i class="pi pi-search-line pi-1x" />
+              </div>
+            </template>
+          </Input>
+        </td>
+      </tr>
+    </thead>
+
+    <tbody class="table-body">
+      <tr v-for="(row, rowIndex) in sortedColumns" :key="rowIndex">
+        <td
+          v-for="(column, columnIndex) in columns"
+          :key="columnIndex"
+          :align="column.align || 'start'"
+          class="table-cell"
+        >
+          <slot name="item" :item="row">
+            <slot :name="'item-' + Object.keys(row)[columnIndex]" :item="row">
+              {{ row[Object.keys(row)[columnIndex]] }}
+            </slot>
+          </slot>
+        </td>
+      </tr>
+
+      <tr>
+        <td
+          v-if="sortedColumns.length === 0"
+          :colspan="Object.keys(columns).length"
+          align="center"
+          class="no-search-results"
+        >
+          <div>No results found</div>
+          <Button @click="clearSearch" miter width="170px" height="36px"
+            >Clear search</Button
+          >
+        </td>
+      </tr>
+    </tbody>
+  </table>
+</template>
+
+<script lang="ts">
+import { defineComponent } from 'vue'
+import Button from '../Button/Button.vue'
+import Input from '../Input/Input.vue'
+
+export default defineComponent({
+  name: 'Datatable',
```

Comment:
We'll need to update the filename as well

```suggestion
  name: 'DataTable',
```